rickshaw-run

```diff
@@ -1825,73 +1976,27 @@ sub prepare_bench_tool_engines() {
-                    $src_file =~ s/\%run-dir\%/$run_dir\//g;
-                    $src_file =~ s/\%config-dir\%/$config_dir\//g;
-                    my $dest_file = $$file_spec{'dest'};
-                    if (-e $src_file) {
-                        debug_log(sprintf("adding %s for tool '%s'", $src_file, $tool_name));
-                        printf FH "src=%s\ndest=%s\n", $src_file, $dest_file;
-                    } elsif (! $$file_spec{'required'}) {
-                        debug_log(sprintf("skipping %s because it does not exist and is not required for tool '%s'", $src_file, $tool_name));
-                    } else {
-                        die "[ERROR]Could not find required file " . $src_file . " for tool '" . $tool_name . "'\n";
-                    }
-                }
+        if ($cs_type =~ /^client|server|profiler?/) {
+            foreach my $cs_ref (@{ $clients_servers{$cs_type} }) {
+                my $cs_id = $$cs_ref{'id'};
+                my $benchmark = $ids_to_benchmark{$cs_id};
+                build_files_list($cs_type, $cs_id, $benchmark);
             }
+        } else {
+            build_files_list($cs_type);
         }
-        foreach my $utility (@utilities) {
-            my $utility_dir = $run{$utility . '-dir'};
-            if (exists $utility_configs{$utility}{'engine'}{'files-from-controller'}) {
-                for my $file_spec (@{ $utility_configs{$utility}{'engine'}{'files-from-controller'} }) {
-                    if (! $$file_spec{'required'}) {
-                        $$file_spec{'required'} = 1;
-                    }
-                    my $src_file = $$file_spec{'src'};
-                    $src_file =~ s/\%utility-dir\%/$utility_dir/g;
-                    my $dest_file = $$file_spec{'dest'};
-                    if (-e $src_file) {
-                        debug_log(sprintf("adding %s for utility '%s'", $src_file, $utility));
-                        printf FH "src=%s\ndest=%s\n", $src_file, $dest_file;
-                    } elsif (! $$file_spec{'required'}) {
-                        debug_log(sprintf("skipping %s because it does not exist and is not required for utility '%s'", $src_file, $utility));
-                    } else {
-                        die "[ERROR]Could not find required file " . $src_file . " for utility '" . $utility . "'\n";
-                    }
-                }
+    }
+
+    # Create empty files for any not created above
+    foreach my $cs_type (keys %clients_servers, @all_collector_types) {
+        foreach my $cs_ref (@{ $clients_servers{$cs_type} }) {
+            my $cs_id = $$cs_ref{'id'};
+            my $cs_file_list = $engine_config_dir . "/" . $cs_type . "-" . $cs_id . "-files-list";
+            if ( ! -e $cs_file_list) {
+                open(FH, ">" . $cs_file_list) || die "[ERROR]could not open " . $cs_file_list . " for writing";
+                close FH;
```

Comment:
I just removed this :)  I think this is a crutch we should get rid of.  If we are not creating files above I think we should let it error so that we can figure out why and fix it.

Reply:
ok, should be fine to remove.